src/main/fc/fc_rc.c

```diff
@@ -169,6 +169,8 @@ static void calculateSetpointRate(int axis)
 
         angleRate = applyRates(axis, rcCommandf, rcCommandfAbs);
     }
+    // Rate limit from profile (deg/sec)
+    angleRate = constrainf(angleRate, -1.0f * currentControlRateProfile->rate_limit[axis], 1.0f * currentControlRateProfile->rate_limit[axis]);
```

Comment:
This is inefficient. We should make sure that `currentControlRateProfile->rate_limit[axis]` does not exceed `SETPOINT_RATE_LIMIT` (probably best in `validateAndFixConfig()` in `src/main/fc/config.c` instead of running 2 `constrainf()` at run time.

Reply:
How best should I pass the local `#define SETPOINT_RATE_LIMIT 1998` out so it can be used to validate the `currentControlRateProfile->rate_limit[axis]`? It's not in any header I can include.
I assume it's local to the source for maximum robustness (also used by `scaleRcCommandToFpvCamAngle()` so as to prevent the trigonometric adjusted rates from exceeding hardware limits).
Have pushed a refactored version of the concept with `CONTROL_RATE_CONFIG_RATE_LIMIT_MIN` and `CONTROL_RATE_CONFIG_RATE_LIMIT_MAX` moved to `rc_controls.h` for global use.
As a general question, how robust are the limits enforced by `settings.c`? None of the other 'controlRateConfig_t' members receive validation in `validateAndFixConfig()`. If they are not considered robust then maybe the runtime penalty of 2 `constrainf()` calls is justified?
Also as an open query, should the structure member initialisation in `controlrate_profile.c` in `pgResetFn_controlRateProfiles()` use #defines when available (as I have done), or hard-coded numbers?